examples/step-60/step-60.cc

```diff
@@ -0,0 +1,1077 @@
+// description of your domain. In this case, the description is done through the
+// actual *deformation* field, rather than a *displacement* field.
+//
+// Which one is used depends on how the user wants to specify the reference
+// domain, and/or the actual configuration. We'll provide both options, and
+// experiment a little in the results section of this tutorial program.
+
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/base/parsed_function.h>
+
+// The parsed function class is another new entry. It allows one to create a
+// Function object, starting from a string in a parameter file which is parsed
+// into an object that you can use anywhere deal.II accepts a Function (for
+// example, for interpolation, boundary conditions, etc.).
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/matrix_tools.h>
+
+#include <deal.II/non_matching/coupling.h>
+
+// This is the last new entry for this tutorial program. The namespace
+// NonMatching contains a few methods that are useful when performing
+// computations on non-matching grids, or on curves that are not aligned with
+// the underlying mesh.
+//
+// We'll discuss its use in details later on in the `setup_coupling` method.
+
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/sparse_direct.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/linear_operator.h>
+#include <deal.II/lac/linear_operator_tools.h>
+
+#include <iostream>
+#include <fstream>
+
+namespace Step60
+{
+  using namespace dealii;
+
+  // @sect3{DistributedLagrangeProblem}
+  //
+  // In the DistributedLagrangeProblem, we need two parameters describing the
+  // dimensions of the domain $\Gamma$ (`dim`) and of the domain $\Omega$
+  // (`spacedim`).
+  //
+  // These will be used to initialize a Triangulation<dim,spacedim> (for
+  // $\Gamma$) and a Triangulation<spacedim,spacedim> (for $\Omega$).
+  //
+  // A novelty w.r.t. other tutorial programs is the heavy use of
+  // std::unique_ptr. These behave like classical pointers, with the advantage
+  // of doing automatic house-keeping: the contained object is automatically
+  // destroyed as soon as the unique_ptr goes out of scope, even if it is inside
+  // a container or there's an exception. Moreover it does not allow for
+  // duplicate pointers, which prevents ownership problems. We do this, because
+  // we want to be able to i) construct the problem, ii) read the parameters,
+  // and iii) initialize all objects according to what is specified in a
+  // parameter file.
+  //
+  // We construct the parameters of our problem in the internal class
+  // DistributedLagrangeProblemParameters, derived from ParameterAcceptor. The
+  // DistributedLagrangeProblem class takes a const reference to a
+  // DistributedLagrangeProblemParameters object, so that it is not possible to
```

Comment:
Since the parameters class is already within the `DistributedLagrangeProblem` class, I'd just call it `DistributedLagrangeProblem::Parameters`.